### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by Laylo309**
- Use preprocessed HTML file with embedded Ruby code.
- Use layouts and templates for shared content.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the current_user.
    - Create a comment on behalf of the current_user.
- No linter errors😎

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,4 +9,25 @@ def show
...
+        redirect_to user_post_path(@post.user.id, @post.id) if @post.save
+      end
+    end
+  end
+
+  private
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```
Comment (Dipeshtwis) [inline app/controllers/posts_controller.rb]: - [x] `post_params` never used here. Please make sure to use the correct params while creating a post.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,3 +1,18 @@
 class CommentsController < ApplicationController
-  def comments; end
+  def create
+    post = Post.find(params[:post_id])
+    comment = post.comments.new(text: comment_params[:text], user: current_user)
+
+    respond_to do |format|
```
Comment (Dipeshtwis) [inline app/controllers/comments_controller.rb:6]: - [x] The comments controller should have new actions similar to the one shown below. The `new` is used with the HTTP verb GET and creates an object instance.

```ruby

def new
  @comment = Comment.new
end
```

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
-  get 'likes/likes'
-  get 'comments/comments'
-  resources :users, only: [:index, :show] do resources :posts, only: [:index, :show] end
+  resources :users, only: [:index, :show] do resources :posts, only: [:index, :show, :new] end
+
+    resources :posts, only: [:new, :create] do
+    resources :comments, only: [:create]
+    resources :likes, only: [:create]
```
Comment (Dipeshtwis) [inline config/routes.rb]: - [x] Also add `new` route for comments.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Laylo309, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Laylo309) [reply app/controllers/posts_controller.rb]: Done ✅

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (Laylo309) [reply app/controllers/comments_controller.rb:6]: Done ✅

File: config/routes.rb (hunk 3 above)
Comment (Laylo309) [reply config/routes.rb]: Done ✅

**Comment by Davidosky007** [comment]
## Hi @Laylo309 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 


CR GIF

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**